diff --git a/node_modules/react-native-quick-crypto/lib/commonjs/ec.js b/node_modules/react-native-quick-crypto/lib/commonjs/ec.js
index 26c1553..0aea59e 100644
--- a/node_modules/react-native-quick-crypto/lib/commonjs/ec.js
+++ b/node_modules/react-native-quick-crypto/lib/commonjs/ec.js
@@ -356,7 +356,7 @@ function ecDeriveBits(algorithm, baseKey, length) {
   // If length is specified, truncate
   const byteLength = Math.ceil(length / 8);
   if (secretBuf.byteLength >= byteLength) {
-    return secretBuf.subarray(0, byteLength).buffer;
+    return secretBuf.buffer.slice(secretBuf.byteOffset, secretBuf.byteOffset + byteLength);
   }
   throw new Error('Derived key is shorter than requested length');
 }
diff --git a/node_modules/react-native-quick-crypto/lib/module/ec.js b/node_modules/react-native-quick-crypto/lib/module/ec.js
index 3f37c06..d412f0b 100644
--- a/node_modules/react-native-quick-crypto/lib/module/ec.js
+++ b/node_modules/react-native-quick-crypto/lib/module/ec.js
@@ -344,7 +344,7 @@ export function ecDeriveBits(algorithm, baseKey, length) {
   // If length is specified, truncate
   const byteLength = Math.ceil(length / 8);
   if (secretBuf.byteLength >= byteLength) {
-    return secretBuf.subarray(0, byteLength).buffer;
+    return secretBuf.buffer.slice(secretBuf.byteOffset, secretBuf.byteOffset + byteLength);
   }
   throw new Error('Derived key is shorter than requested length');
 }
diff --git a/node_modules/react-native-quick-crypto/src/ec.ts b/node_modules/react-native-quick-crypto/src/ec.ts
index 49b4b4b..68c3ab7 100644
--- a/node_modules/react-native-quick-crypto/src/ec.ts
+++ b/node_modules/react-native-quick-crypto/src/ec.ts
@@ -550,7 +550,7 @@ export function ecDeriveBits(
   // If length is specified, truncate
   const byteLength = Math.ceil(length / 8);
   if (secretBuf.byteLength >= byteLength) {
-    return secretBuf.subarray(0, byteLength).buffer as ArrayBuffer;
+    return secretBuf.buffer.slice(secretBuf.byteOffset, secretBuf.byteOffset + byteLength) as ArrayBuffer;
   }
 
   throw new Error('Derived key is shorter than requested length');
